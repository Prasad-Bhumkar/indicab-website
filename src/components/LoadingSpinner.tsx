import React from 'react';

const _LoadingSpinner = (): JSX.Element => {
    return (
        <div className="flex justify-center items-center p-4">
            <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-gray-900"></div>
        </div>
    );
};

export default _LoadingSpinner;
