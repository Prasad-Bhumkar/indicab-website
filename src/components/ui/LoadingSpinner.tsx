import React from "react";

"use client";

interface LoadingSpinnerProps {
    className?: string;
}

export default function LoadingSpinner({ className = '' }: LoadingSpinnerProps): JSX.Element {
    return (
        <div className={`flex justify-center items-center py-20 ${className}`}>
            <div className="h-10 w-10 border-4 border-primary border-t-transparent rounded-full animate-spin"></div>
        </div>
    );
}
