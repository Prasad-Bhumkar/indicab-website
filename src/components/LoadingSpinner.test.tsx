import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner Component', () => {
    it('renders spinner element', () => {
        render(<LoadingSpinner />);
        expect(screen.getByRole('status')).toBeInTheDocument();
    });

    it('contains loading text', () => {
        render(<LoadingSpinner />);
        expect(screen.getByText(/loading/i)).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { container } = render(<LoadingSpinner />);
        expect(container).toMatchSnapshot();
    });
});
