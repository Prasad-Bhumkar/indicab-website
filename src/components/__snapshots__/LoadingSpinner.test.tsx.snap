// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner Component > matches snapshot 1`] = `
<div>
  <div
    class="flex justify-center items-center p-4"
  >
    <div
      class="animate-spin rounded-full h-8 w-8 border-b-2 border-gray-900"
    />
  </div>
</div>
`;
